dnf/base.py

```diff
@@ -212,6 +228,100 @@ def _setup_excludes_includes(self, only_main=False):
             for query, repoid in repo_excludes:
                 self.sack.add_excludes(query)
 
+    def _setup_modules(self):
+        module_defaults_prioritizer = Modulemd.Prioritizer()
+
+        # read defaults from repos
+        for repo in self.repos.iter_enabled():
+            try:
+                module_metadata = ModuleMetadataLoader(repo).load()
+
+                defaults = [i for i in module_metadata if isinstance(i, Modulemd.Defaults)]
+                module_defaults_prioritizer.add(defaults, 0)
+
+                # read modules from modulemd
+                for data in module_metadata:
+                    if not isinstance(data, Modulemd.Module):
+                        continue
+                    self.repo_module_dict.add(RepoModuleVersion(data, base=self, repo=repo))
+            except (dnf.exceptions.Error, GLib.Error) as e:
+                logger.debug(e)
+                continue
+
+        # read defaults from disk
+        # chroot behavior:
+        #   * use host configuration (repos work the same)
+        #   * to use chroot configuration, you need to switch to chroot
+        topdir = self.conf.moduledefaultsdir._get()
+        try:
+            for fn in os.listdir(topdir):
+                if not fn.endswith(".yaml"):
+                    continue
+                yaml_file = os.path.join(topdir, fn)
+
+                with open(yaml_file, "r") as f:
```

Comment:
Not fixed yet - Test if path is file or readable file could help. If it will be a directory there will be a lot of fun.